chains/substrate/pallet/pallet.go

```diff
@@ -53,15 +64,27 @@ func (p *Pallet) ExecuteProposals(
 		})
 	}
 
-	return p.Transact(
-		"SygmaBridge.execute_proposal",
-		bridgeProposals,
-		signature,
-	)
+	if p.isCheckMetadataModeEnabled {
```

Comment:
Would prefer if we don't do this and create an interface in the constructor and pass the correct client which we choose in the main app.go based on the network type.

Reply:
I agree that this approach would be much better in this case